core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1874,7 +1875,23 @@ private RexNode convertOver(Blackboard bb, SqlNode node) {
               RexWindowBound.create(window.getLowerBound(), lowerBound),
               RexWindowBound.create(window.getUpperBound(), upperBound),
               window);
-      return rexAgg.accept(visitor);
+      //set it in a recursive method?
+      RexNode overNode = rexAgg.accept(visitor);
+      if (aggCall.getFunctionQuantifier() != null
+        && aggCall.getFunctionQuantifier().getValue().equals(SqlSelectKeyword.DISTINCT)) {
+        ImmutableList<RexNode> list = ((RexCall) overNode).operands.asList();
+        for (int i = 0; i < list.size(); i++) {
+          if (list.get(i) instanceof RexCall) {
+            ImmutableList<RexNode> listOperands = ((RexCall) list.get(i)).operands;
+            for (int k = 0; k < listOperands.size(); k++) {
+              if (listOperands.get(k) instanceof RexOver) {
+                ((RexOver) listOperands.get(k)).setDistinct(true);
```

Comment:
@julianhyde Here we are actually making the bridge between the SQLBasicCall where we have the distinct aggregate and the RexOver node where distinct field was added. I am not sure if this is the best way to set the distinct field for all aggregates belonging to this particular call. If you have a better suggestion please let me know